src/codecs/json.rs

```diff
@@ -0,0 +1,167 @@
+use super::{Codec, CodecTransform};
+use crate::{
+    config::DataType,
+    transforms::json_parser::{JsonParser, JsonParserConfig},
+};
+use serde::{Deserialize, Serialize};
+use vector_core::{
+    config::log_schema,
+    event::{Event, Value},
+    transform::{FunctionTransform, Transform},
+};
+
+#[derive(Debug, Clone, Serialize, Deserialize, Derivative)]
+#[serde(deny_unknown_fields, default)]
+#[derivative(Default)]
+pub struct JsonCodec {
+    mode: Option<Mode>,
```

Comment:
This doesn't seem to be used for anything.

Reply:
Oops, yes I accidentally left that in from the previous (admittedly weird) feature: https://github.com/timberio/vector/pull/7803/commits/f4e290f23e2d5f2aaacda3e39d1640d44f114b45#diff-61a423b9da07f268a42699f3bf330341a5a596b16439332c9f98b704c776ae73R104-R109